### PR TITLE
Restructure aws signing round tripper

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

- Move RoundTripper for signing aws requests to awsauth package
- Remove NewClientWithAwsSigner function and replace with generic NewClientWithTransport that passes the roundtripper into the client transport layer. This will allow future use of this method for various niche transport layers needed by our apps.
- Additional unit test to cover SharedCredentials method

### How to review

Check changes make sense

### Who can review

Anyone

### CODE REVIEW

File: awsauth/roundtripper.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	if awsRegion == "" || awsService == "" {
+		return nil, fmt.Errorf("aws region and service should be valid options")
+	}
+
+	awsSigner, err := NewAwsSigner(awsFilename, awsProfile, awsRegion, awsService)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create aws v4 signer: %w", err)
+	}
+
+	if len(options) > 0 && options[0].TlsInsecureSkipVerify {
```
Comment (rahulPalliyalil) [inline awsauth/roundtripper.go:42]: (question): do we actually need to provide ```TlsInsecureSkipVerify``` option to the consumer ? I mean I am not sure whether we should allow any certificate that comes from the server, we should only allow domain specific certificate right ?

File: awsauth/roundtripper_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	dphttp "github.com/ONSdigital/dp-net/v2/http"
+	"github.com/ONSdigital/dp-net/v2/http/httptest"
+	. "github.com/smartystreets/goconvey/convey"
```
Comment (rahulPalliyalil) [inline awsauth/roundtripper_test.go:12]: I guess we don't need to prefix this import with a dot ?

File: awsauth/roundtripper_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	t.Parallel()
+
+	awsSignerRT, err := NewAWSSignerRoundTripper("some_filename", "", "some_region", "")
+
+	assert.NotNil(t, err, "error should not be nil")
+	assert.Nil(t, awsSignerRT, "aws signer roundtripper should return nil")
+}
+
+func TestNewClientWithTransport(t *testing.T) {
+	Convey("Given a access and secret key are set in the environement", t, func() {
```
Comment (rahulPalliyalil) [inline awsauth/roundtripper_test.go:44]: can't these be run in parallel as well ?

File: ci/lint.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.16
+    tag: latest
```
Comment (rahulPalliyalil) [inline ci/lint.yml:9]: i guess we need to fix the go version wherever possible, specifically for the linter . As if you tag it with latest, then some linters in golangcilint won't be forward compatible with unexpected version update and we will start experiencing unexpected   lint issues in the pipeline.

File: http/client_test.go [hunk 5]
```diff
@@ -373,6 +373,37 @@ func TestSetPathsWithNoRetries(t *testing.T) {
 	})
 }
 
+func TestNewClientWithTransport(t *testing.T) {
+	Convey("Given a custom http transport", t, func() {
```
Comment (rahulPalliyalil) [inline http/client_test.go:377]: wouldn't it be possible to run these in parallel like others ?

**Review by rahulPalliyalil — CHANGES_REQUESTED** [review]

File: awsauth/roundtripper.go (hunk 1 above)
Comment (nshumoogum) [reply awsauth/roundtripper.go:42]: This is for developer use, to be able to connect from localhost to Elasticsearch cluster

Comment (rahulPalliyalil) [reply awsauth/roundtripper.go:42]: in that case i guess we need to control the activation of this using an environment variable right ?

Comment (nshumoogum) [reply awsauth/roundtripper.go:42]: yep - that is what i've done on my branch in dp-search-api, and also made a new make target local

Comment (nshumoogum) [reply awsauth/roundtripper.go:42]: I could update the docs in awsauth to make it clear this should only be used when testing locally

File: awsauth/roundtripper_test.go (hunk 2 above)
Comment (nshumoogum) [reply awsauth/roundtripper_test.go:12]: We do

**Review by rahulPalliyalil — APPROVED** [review]

**Pull request merged**